ui/src/InvalidAddress.js

```diff
@@ -0,0 +1,107 @@
+import React, {Component} from 'react';
+import '@fontsource/roboto';
+import {Button, Dialog, DialogContent, TextField} from '@material-ui/core';
+
+class InvalidAddress extends Component {
+  state = {
+    reason: '',
+    notes: '',
+    reasonValidationError: false,
+    showDialog: false,
+  }
+
+  openDialog = () => {
+    this.setState({
+      showDialog: true,
+    })
+  }
+
+  closeDialog = () => {
+    this.setState({
+      reason: '',
+      notes: '',
+      reasonValidationError: false,
+      showDialog: false,
+    })
+  }
+
+  onReasonChange = (event) => {
+    this.setState({
+      reason: event.target.value,
+      reasonValidationError: false
+    })
+  }
+
+  onNotesChange = (event) => {
+    this.setState({
+      notes: event.target.value
+    })
+  }
+
+  onCreate = async () => {
+    if (!this.state.reason) {
+      this.setState({reasonValidationError: true})
+
+      return
+    }
+
+    const invalidAddress = {
+      "reason": this.state.reason,
+      "notes": this.state.notes,
+    }
+
+    const response = await fetch(`/cases/${this.props.caseId}/action/invalid-address`, {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(invalidAddress)
+    })
+
+    if (response.ok) {
+      this.closeDialog()
+    }
+  }
+
+  render() {
+    return (
+        <div>
+          <Button
+              style={{marginTop: 10, marginBottom: 10}}
+              onClick={this.openDialog}
+              variant="contained">
+            Invalidate this case
+          </Button>
+          <Dialog open={this.state.showDialog}>
+            <DialogContent style={{padding: 30}}>
+              <div>
+                <TextField
+                    required
+                    inputProps={{style: {textTransform: 'uppercase'}}}
```

Comment:
why are you forcing the reason to be uppercase?

Reply:
Ah, I thought we used to get the reason as an enumerated value for a while on census before (I forgot!) it was switched to free text. Will remove this